packages/worker/src/api/controllers/system/tenants.ts

```diff
@@ -5,7 +5,7 @@ export async function destroy(ctx: UserCtx) {
   const user = ctx.user!
   const tenantId = ctx.params.tenantId
 
-  if (tenantId !== user.tenantId) {
+  if (!ctx.internal && tenantId !== user.tenantId) {
```

Comment:
Are we using user tokens for internal calls?

Reply:
we use the `INTERNAL_API_KEY`

Reply:
In this case we want to only allow the action if the current session matches the users tenant id, or the api key is the internal api (system) api key